run-ib-pr-matrix.sh

```diff
@@ -32,6 +32,15 @@ if [ "${CHECK_WORKFLOWS}" = "true" ] ; then
     WFS="${wf},${WFS}"
   done
   WFS=$(echo ${WFS} | sed 's|,$||')
+
+  set +e
+  check_invalid_wf_lists "${WORKFLOWS}" false
+  if [ $? -ne 0 ]; then
+    WFS=""
+    cat $WORKSPACE/bad-workflow-lists.txt >> ${WORKSPACE}/workflows-${BUILD_ID}.log
+  fi
+  set -e
+
```

Comment:
@iarspider , what is this code doing ? why do we need it for `CHECK_WORKFLOWS=true`

Reply:
check_invalid_wf_lists will return 1 if all requested workflows are invalid (i.e. passing them to runTheMatrix will make it run every single workflow). We need it for `CHECK_WORKFLOWS=true` to not trigger baseline job with every single workflow. We need it for `CHECK_WORKFLOWS=false` as well (line 90) to avoid running every single workflow during PR testing.

Reply:
ah ok.
I think `workflows-${BUILD_ID}.log` is a special file which we use to decide it baseline should be triggered ( see https://cmssdt.cern.ch/jenkins/job/ib-check-baseline-requests/configure ), so better to not override the contents of it with bad-workflow-lists.txt

Reply:
> better to not override the contents of it with bad-workflow-lists.txt

Noted. Will remove that line.